src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -655,6 +655,16 @@ private static IQueryMatch HandleVector(Parameters builderParameters, MethodExpr
         {
             transformedEmbedding = GenerateEmbeddings.FromBase64Array(vectorOptions, allocator: builderParameters.Allocator, stringSegment.ToString());
         }
+        else if (value is BlittableJsonReaderObject json)
+        {
+            json.TryGetMember(Sparrow.Global.Constants.Naming.VectorPropertyName, out var vectorObject);
```

Comment:
Can it return `false`? Shouldn't we check that? Even throw explicitly then, exactly as we do below

Reply:
Added proper exception for such case.